petclinic/actors/customers/src/lib.rs

```diff
@@ -1,118 +1,96 @@
-                last_name: Some("PetLover".to_string()),
-                email: "bob@lovespets.com".to_string(),
-                id: 1,
-                ..Default::default()
-            },
-            Owner {
-                first_name: "Alice".to_string(),
-                last_name: Some("PetLover".to_string()),
-                email: "alice@lovespets.com".to_string(),
-                id: 2,
-                ..Default::default()
-            },
-        ])
+    async fn list_owners(&self, ctx: &Context) -> RpcResult<OwnersList> {
+        let db = SqlDbSender::new();
+        let owners = db::list_all_owners(&ctx, &db).await?;
+        Ok(owners.iter().cloned().map(|o| o.into()).collect())
     }
 
-    async fn update_owner(&self, _ctx: &Context, _arg: &Owner) -> RpcResult<UpdateOwnerReply> {
-        Ok(UpdateOwnerReply { success: true })
+    async fn update_owner(&self, ctx: &Context, arg: &Owner) -> RpcResult<UpdateOwnerReply> {
+        let db = SqlDbSender::new();
+        Ok(match db::update_owner(&ctx, &db, arg).await {
+            Ok(_) => UpdateOwnerReply { success: true },
+            Err(_e) => UpdateOwnerReply { success: false },
+        })
     }
 
-    async fn list_pet_types(&self, _ctx: &Context) -> RpcResult<PetTypeList> {
-        Ok(vec![
-            PetType {
-                id: 1,
-                name: "Cat".to_string(),
-            },
-            PetType {
-                id: 2,
-                name: "Dog".to_string(),
-            },
-            PetType {
-                id: 3,
-                name: "Anaconda".to_string(),
-            },
-        ])
+    async fn list_pet_types(&self, ctx: &Context) -> RpcResult<PetTypeList> {
+        let db = SqlDbSender::new();
+        let pettypes = db::list_all_pet_types(&ctx, &db).await?;
+        Ok(pettypes.iter().cloned().map(|o| o.into()).collect())
     }
 
-    async fn add_pet(&self, _ctx: &Context, _arg: &AddPetRequest) -> RpcResult<bool> {
-        Ok(true)
+    async fn add_pet(&self, ctx: &Context, arg: &AddPetRequest) -> RpcResult<bool> {
+        let db = SqlDbSender::new();
+        Ok(match db::add_pet(&ctx, &db, arg.owner_id, &arg.pet).await {
+            Ok(_) => true,
+            Err(_e) => false,
+        })
     }
 
-    async fn remove_pet(&self, _ctx: &Context, _arg: &u64) -> RpcResult<bool> {
-        Ok(true)
+    async fn remove_pet(&self, ctx: &Context, arg: &u64) -> RpcResult<bool> {
+        let db = SqlDbSender::new();
+        Ok(match db::delete_pet(&ctx, &db, *arg).await {
+            Ok(_) => true,
+            Err(_e) => false,
+        })
     }
 
-    async fn update_pet(&self, _ctx: &Context, _arg: &Pet) -> RpcResult<bool> {
-        Ok(true)
+    async fn update_pet(&self, ctx: &Context, arg: &Pet) -> RpcResult<bool> {
+        let db = SqlDbSender::new();
+        Ok(match db::update_pet(&ctx, &db, arg).await {
+            Ok(_) => true,
+            Err(_e) => false,
+        })
```

Comment:
If you want to turn a result into a boolean, you can just use is_ok(), so
 `Ok(db::update_pet(&ctx,&db,arg).await.is_ok())`

Reply:
Yeah I left the match because I'm likely going to drop a call to error! in the fail arm.